src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -241,6 +246,39 @@ impl StreamSink {
             }
         }
     }
+
+    /// The table schema is: | epoch | seq id | row op | sink columns |
+    /// Pk is: | epoch | seq id |
+    fn infer_kv_log_store_table_catalog(&self) -> TableCatalog {
+        let mut table_catalog_builder =
+            TableCatalogBuilder::new(self.input.ctx().with_options().internal_table_subset());
+
+        // There are three pre-defined columns in the table: `epoch`, `seq_id` and `row_op`.
+        table_catalog_builder.add_column(&Field::with_name(EPOCH_COLUMN_TYPE, EPOCH_COLUMN_NAME));
+        table_catalog_builder.add_column(&Field::with_name(SEQ_ID_COLUMN_TYPE, SEQ_ID_COLUMN_NAME));
+        table_catalog_builder.add_column(&Field::with_name(ROW_OP_COLUMN_TYPE, ROW_OP_COLUMN_NAME));
+        let predefined_column_len = table_catalog_builder.columns().len();
+
+        // The table's pk is composed of `epoch` and `seq_id`.
+        table_catalog_builder.add_order_column(EPOCH_COLUMN_INDEX, OrderType::ascending());
+        table_catalog_builder
+            .add_order_column(SEQ_ID_COLUMN_INDEX, OrderType::ascending_nulls_last());
+        let read_prefix_len_hint = table_catalog_builder.get_current_pk_len();
+
+        let value_indices = table_catalog_builder.extend_columns(&self.sink_desc().columns);
```

Comment:
I just tried running with kv log store. It fails because the value_indices of predefined columns are not included here. 

We should fix it before merging into main.

Reply:
Fixed.